src/SonarScanner.MSBuild.Tasks/IsTestByReference.cs

```diff
@@ -0,0 +1,95 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace SonarScanner.MSBuild.Tasks
+{
+    /// <summary>
+    /// MSBuild task that determines whether the project should be treated as a test project for analysis purposes based on its references.
+    /// </summary>
+    public sealed class IsTestByReference : Task
+    {
+        private static readonly ISet<string> TestAssemblyNames = new HashSet<string>
```

Comment:
could you add a test that verifies the number of assembly names in this set? (and make this internal for the test, I imagine)

The UT should fail and give a message to update sonar-dotnet and sonar-security C# FE when this list gets updated. Having a test will be complementary to the comment in this class (which can be overlooked).

And we should have similar UTs in the other code bases. I think it's ok to feel some pain when updating the list, in order to make sure we don't forget about the duplication.

Reply:
Nice idea!